src/cli/rustup_mode.rs

```diff
@@ -55,17 +55,44 @@ where
 pub fn main() -> Result<utils::ExitCode> {
     self_update::cleanup_self_updater()?;
 
+    use clap::ErrorKind::*;
     let matches = match cli().get_matches_from_safe(process().args_os()) {
         Ok(matches) => Ok(matches),
-        Err(e)
-            if e.kind == clap::ErrorKind::HelpDisplayed
-                || e.kind == clap::ErrorKind::VersionDisplayed =>
-        {
-            writeln!(process().stdout().lock(), "{}", e.message)?;
+        Err(clap::Error {
+            kind: HelpDisplayed,
+            message,
+            ..
+        }) => {
+            writeln!(process().stdout().lock(), "{}", message)?;
+            return Ok(utils::ExitCode(0));
+        }
+        Err(clap::Error {
+            kind: VersionDisplayed,
+            message,
+            ..
+        }) => {
+            writeln!(process().stdout().lock(), "{}", message)?;
+            info!("This is the version for the rustup toolchain manager, not the rustc compiler.");
+
+            fn rustc_version() -> std::result::Result<String, Box<dyn std::error::Error>> {
+                let cfg = &mut common::set_globals(false, true)?;
+                let cwd = std::env::current_dir()?;
+                let toolchain = cfg.find_or_install_override_toolchain_or_default(&cwd)?.0;
```

Comment:
Unless I'm mistaken, I think this may need to take into account if a `+toolchain` was specified. but I'm not in a position to check right now,